scala-api/src/test/scala/com/cloudera/livy/ScalaClientTest.scala

```diff
@@ -0,0 +1,282 @@
+        client.stop(true)
+      }
+    }
+  }
+
+  test("test Sync Rpc") {
+    configureClient(true)
+    try {
+      pingJob()
+      val future = client.run(helloJob())
+      val result = Await.result(future, 10 second)
+      assert(result === "hello")
+    } catch {
+      case e: Exception => throw e
+    } finally {
+      if (client != null) {
+        client.shutdown()
+        client.stop(true)
+      }
+    }
+  }
+
+  test("test Remote client") {
+    configureClient(false)
+    try {
+      pingJob()
+      val sFuture = client.submit(simpleSparkJob())
+      val result = Await.result(sFuture, 10 second)
+    } catch {
+      case e: Exception => throw e
+    } finally {
+      if (client != null) {
+        client.shutdown()
+        client.stop(true)
+      }
+    }
+  }
+
+  test("test add file") {
+    configureClient(true)
+    var file: File = null
+    try {
+      pingJob()
+      file = File.createTempFile("test", ".file")
+      val fileStream = new FileOutputStream(file)
+      fileStream.write("test file".getBytes("UTF-8"))
+      fileStream.close
+      val future = client.addFile(new URI("file:" + file.getAbsolutePath()))
+      Thread.sleep(5000)
+      val sFuture = client.submit(
+        context => ScalaClientTest.fileOperation(false, file.getName, context)
+      )
+      val output = Await.result(sFuture, 10 second)
+      assert(output === "test file")
+    } finally {
+      if (client != null) {
+        client.shutdown()
+        client.stop(true)
+      }
+      if (file != null) {
+        file.delete()
+      }
+    }
+  }
+
+  test("test add jar") {
+    configureClient(true)
+    var jar: File = null
+    try {
+      pingJob()
+      jar = File.createTempFile("test", ".resource")
+      var jarFile = new JarOutputStream(new FileOutputStream(jar))
+      jarFile.putNextEntry(new ZipEntry("test.resource"))
+      jarFile.write("test resource".getBytes("UTF-8"))
+      jarFile.closeEntry()
+      jarFile.close()
+      val future = client.addJar(new URI("file:" + jar.getAbsolutePath()))
+      Thread.sleep(5000)
+      val sFuture = client.submit(
+        context => ScalaClientTest.fileOperation(true, "test.resource", context)
+      )
+      val output = Await.result(sFuture, 10 second)
+      assert(output === "test resource")
+    } finally {
+      if (client != null) {
+        client.shutdown()
+        client.stop(true)
+      }
+      if (jar != null) {
+        jar.delete()
+      }
+    }
+  }
+
+  private def configureClient(local: Boolean) = {
+    var conf = ScalaClientTest.createConf(local)
+    var javaClient = new LivyClientBuilder(false).setURI(new URI("rsc:/")).setAll(conf).build()
+    client = javaClient.asScalaClient
+  }
+
+  private def pingJob() = {
+    val future = client.submit(context => {
+      null
+    })
+    val result = Await.result(future, 5 second)
+    assert(result == null)
+  }
+
+  private def helloJob(): (ScalaJobContext) => String = {
```

Comment:
Instead of this, just declare:

    private def hello(jc: ScalaJobContext): String = "hello"

And then:

    client.submit(hello)